Packs/Thycotic/Integrations/Thycotic/Thycotic.py

```diff
@@ -0,0 +1,132 @@
+import demistomock as demisto
+from CommonServerPython import *
+from CommonServerUserPython import *
+
+# import json
+import requests
+# import dateparser
+# import time
+# from datetime import datetime, timedelta
+
+# Disable insecure warnings
+requests.packages.urllib3.disable_warnings()
+
+''' CONSTANTS '''
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'
+
+
+class Client(BaseClient):
+    """
+    Client will implement the service API, and should not contain any Demisto logic.
+    Should only do requests and return data.
+    """
+    def __init__(self, server_url: str, username: str, password: str, max_fetch: int):
+        super().__init__(base_url=server_url)
+        self._username = username
+        self._password = password
+        self._max_fetch = max_fetch
+        self._token = self._generate_token()
+        self._headers = {'Authorization': self._token, 'Content-Type': 'application/json'}
+
+    def _generate_token(self) -> str:
+        """Generate an Access token using the user name and password
+        :return: valid token
+        """
+        body = {
+            "username": self._username,
+            "password": self._password,
+            "grant_type": "password"
+        }
+
+        headers = {
+            'Content-Type': 'application/x-www-form-urlencoded'
+        }
+        return "Bearer " + (self._http_request("POST", "/oauth2/token", headers=headers, data=body))['access_token']
+
+    def getToken(self):
+        return self._token
+
+    def getPasswordById(self, secret_id: str) -> str:
+        url_suffix = "/api/v1/secrets/" + secret_id + "/fields/password"
+        return self._http_request("GET", url_suffix)
+
+    def getUsernameById(self, secret_id: str) -> str:
+        url_suffix = "/api/v1/secrets/" + secret_id + "/fields/username"
+        return self._http_request("GET", url_suffix)
+
+
+def test_module(client: Client,) -> str:
+    if client._token == '':
+        raise Exception('Dont access token')
```

Comment:
please consider explaining what should the customer do, e.g:
`Generating an access token failed, please make sure your user has the <> role.`